### ISSUE TITLE
No error shown when using LegacyOctalEscapeSequence in string literal with Strict Mode

### ISSUE THREAD

**Issue by kpreisser** [Microsoft/TypeScript#15949]
Hi, sorry if this is a duplicate.
I found that TypeScript is not reporting an error when using _LegacyOctalEscapeSequence_ in a string literal when using Script Mode (according to [ES2015 11.8.4](https://www.ecma-international.org/ecma-262/6.0/#sec-literals-string-literals)):

**TypeScript Version:**  2.3.1-insiders.20170416

**Code**

```ts
function foo() {
    "use strict";    
    var s = "\04";
}
```

**Expected behavior:**
SyntaxError: `Octal literals are not allowed in strict mode.`

![octal-sequence](https://cloud.githubusercontent.com/assets/13289184/26243350/d77b8f12-3c8b-11e7-9f3b-6453b78e3d7d.png)

**Actual behavior:**
No error.

Thanks!

**Comment by mhegazy** [comment]
Duplicate of https://github.com/Microsoft/TypeScript/issues/396. PRs are welcomed for that one by the way.
